policy/bundle.go

```diff
@@ -424,15 +424,102 @@ func (p *Bundle) PolicyMRNs() []string {
 	return mrns
 }
 
-// SortContents of this policy bundle sorts Queries and Policies by MRNs
+// Deprecated: Use SortContentsV2
 func (p *Bundle) SortContents() {
-	sort.SliceStable(p.Queries, func(i, j int) bool {
-		return p.Queries[i].Mrn < p.Queries[j].Mrn
+	err := p.SortContentsV2(SortConfig{
+		SortQueries: &SortField{
+			Mrn: true,
+		},
+		SortPolicies: &SortField{
+			Mrn: true,
+		},
 	})
+	if err != nil {
+		panic(err)
+	}
+}
 
-	sort.SliceStable(p.Policies, func(i, j int) bool {
-		return p.Policies[i].Mrn < p.Policies[j].Mrn
-	})
+type SortConfig struct {
+	SortQueries  *SortField
+	SortPolicies *SortField
+	SortVariants *SortField
+}
+
+type SortField struct {
+	Uid bool
```

Comment:
Can we not use do this automatic, since MRN includes the uids as well, I am not sure why we need both.

Reply:
I added a check to verify if mrn isnt empty. If we're formatting a bundle that has no mrns, we need to fallback to uids